src/lib/y2storage/planned/lvm_lv.rb

```diff
@@ -93,18 +110,40 @@ def initialize_from_real_lv(real_lv)
         self.reuse_name = real_lv.lv_name
       end
 
-      # Returns the size for the logical volume in a given volume group
+      # Returns the size for the logical volume in a given volume group/thin pool
+      #
+      # * If size is specified as a percentage, it calculates the size using
+      #   the container size as reference.
+      # * If it is a thin volume, it returns max size. If it is unlimited,
+      #   the thin pool size will be returned.
+      # * Otherwise, the planned size (Planned::LvmLv#size) is returned.
+      #
+      # @param container [LvmVg,LvmLv] Volume group or thin pool where the
+      #   logical volume will be placed
+      # @return [DiskSize]
+      def size_in(container)
+        return size_in_percentage(container) if percent_size
+        return size_in_thin_pool(container) if lv_type == LvType::THIN
+        size
+      end
+
+      # Returns the real size for the logical volume in a given volume group/thin pool
       #
-      # It returns the planned size (Planned::LvmLv#size) unless a
-      # percentage has been specified. In that case, it will use the volume
-      # group size and Planned::LvmLv#percent_size to calculate the
-      # desired size.
+      # When dealing with thin pools, some space is reserved for metadata. This method
+      # returns an adjusted planned size taking the available space for the given
+      # lv_type into account.
       #
-      # @param volume_group [LvmVg] Volume group where the logical volume will be placed
+      # @param container [LvmVg,LvmLv] Volume group or thin pool where the
+      #   logical volume will be placed
       # @return [DiskSize]
-      def size_in(volume_group)
-        return size unless percent_size
-        (volume_group.size * percent_size / 100).ceil(volume_group.extent_size)
+      def real_size_in(container)
+        [size_in(container), container.max_size_for_lvm_lv(lv_type)].min
+      end
+
+      # It adds a logical volume as a thin pool
```

Comment:
NP: I guess you want to say `It adds a thin logical volume to a thin pool`, isn't it?